src/Modules/Guilds/Hooks/useETHPermissions.ts

```diff
@@ -0,0 +1,41 @@
+import PermissionRegistry from 'contracts/PermissionRegistry.json';
+import { useContractRead } from 'wagmi';
+import { useMemo } from 'react';
+
+interface useGetETHPermissionProps {
+  permissionRegistryAddress: string;
+  from: string;
+  to: string;
+  callType: string;
+  functionSignature: string;
+}
+
+export const useGetETHPermission = ({
+  permissionRegistryAddress,
+  from,
+  to,
+  callType,
+  functionSignature,
+}: useGetETHPermissionProps) => {
+  const { data, ...rest } = useContractRead({
+    addressOrName: permissionRegistryAddress,
+    contractInterface: PermissionRegistry.abi,
+    functionName: 'getETHPermission(address,address,bytes4)',
+    args: [from, to, functionSignature],
+    watch: true,

```

Comment:
We should avoid using the `watch` flag here and use `useEventContract` instead.

You can check this PR for the implementation:
https://github.com/DXgovernance/DAVI/pull/424

Reply:
Talked about creating a separate issue for the above: https://github.com/DXgovernance/DAVI/issues/454